types/queries/add_bookmark.sql

```diff
@@ -0,0 +1,7 @@
+UPDATE dailp_user
+SET bookmarked_documents = CASE
+    WHEN $3 = true AND NOT $2 = ANY(bookmarked_documents) THEN array_append(bookmarked_documents, $2)
+    WHEN $3 = false THEN array_remove(bookmarked_documents, $2)
+    ELSE bookmarked_documents
+  END
+WHERE id = $1;
```

Comment:
This will probably need to be more like

```sql
insert into user_bookmarked_audio
  (user_id, document_id)
  values ($1, $2)
  on conflict do nothing;
```
Then you would write a separate thing to "delete from" etc. where "user_id,document_id" match.